src/components/Canvas/Circle.jsx

```diff
@@ -2,14 +2,18 @@
 
 import { CanvasBuilder } from './CanvasBuilder'
 
-/**
- * @type {import('./CanvasBuilder').DrawFactory}
- */
-function drawAnimatedCircle (canvas) {
+const initialState = {
+  i: 0,
+  delta: 0.05
+}
+
+/** @type {import('./CanvasBuilder').DrawFactory} */
+function drawAnimatedCircle (canvas, { drawState, setDrawState }) {
   let requestId
-  let i = 0
+  let { i, delta } = drawState
 
-  function draw () {
+  /** @type {import('./CanvasBuilder').DrawFunction} */
+  function draw ({ isPaused }) {
```

Comment:
`draw` gets `props` now. These can / will be whatever the parent component passes down.